gnome-initial-setup/pages/account/gis-account-page.c

```diff
@@ -102,6 +105,34 @@ on_validation_changed (gpointer        page_area,
   update_page_validation (page);
 }
 
+
+static void
+hide_password_switch_if_needed (GisAccountPage *page)
+{
+  GKeyFile *conf_file = gis_driver_get_vendor_conf_file (GIS_PAGE (page)->driver);
+  GisAccountPagePrivate *priv = gis_account_page_get_instance_private (page);
+  g_autoptr(GError) error = NULL;
+
+  /* do nothing if the configuration file doesn't exist */
+  if (conf_file == NULL)
+    return;
+
+  /* check the conf file to see if the password switch should be hidden */
+  if (!g_key_file_get_boolean (conf_file, CONFIG_ACCOUNT_GROUP,
+                               CONFIG_ACCOUNT_SHOW_PASSWORD_SWITCH_KEY, &error) &&
+      error == NULL) {
+    gis_account_page_local_show_password_switch (GIS_ACCOUNT_PAGE_LOCAL (priv->page_local),
+                                                 FALSE);
+    return;
+  }
+
+  if (!g_error_matches (error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND) &&
```

Comment:
On a second reading – the logic is wrong if you define `[page.account] show-password-switch = true`. In that case, `!g_key_file_get_boolean(...)` is FALSE; the block above is skipped; and we will crash either here or in the `g_warning` arguments because `error` is `NULL`. Please test all four cases: no key, `true`, `false`, `garbage`. There should be a warning only in the last case.

Reply:
Yeah, apparently I didn't tested the `true` value with the final code. It won't crash in the `g_error_matches` since it returns `FALSE` if error is `NULL`, but it does crash on accessing it in the `g_warning` formatting. I have updated the PR.